### PR TITLE
add two benchmark datasets aime-2024 and math-odyssey

### PR DESCRIPTION
**Pull request by wedu-nvidia**
I prepared two datasets aime-2024 and math-odyssey
please review

### CODE REVIEW

File: datasets/aime-2024/prepare.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import json
```
Comment (Kipok) [inline datasets/aime-2024/prepare.py:1]: License header is missing. Plus if you used some other codebases to get part of this logic, we should reference them (if you wrote it all fully yourself, then just the license)

File: datasets/math-odyssey/prepare.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    for ending in answer_endings:
+        if answer.endswith(ending):
+            answer = answer[: -(len(ending))]
+            break
+    return answer
+
+
+if __name__ == "__main__":
+    data_folder = Path(__file__).absolute().parent
+    original_file = str(data_folder / f"original_test.json")
```
Comment (Kipok) [inline datasets/math-odyssey/prepare.py:42]:
```suggestion
    original_file = str(data_folder / "original_test.json")
```

File: datasets/math-odyssey/prepare.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    data_folder.mkdir(exist_ok=True)
+    output_file = str(data_folder / f"test.jsonl")
```
Comment (Kipok) [inline datasets/math-odyssey/prepare.py:44]:
```suggestion
    output_file = str(data_folder / "test.jsonl")
```

File: datasets/aime-2024/prepare.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        return solution, exact_answer
+    elif choice == 'question':
+        filtered_elements = [ele for ele in elements if not ele.startswith('[asy]')]
+        question = ' '.join(filtered_elements)
+        return question
+
+
+if __name__ == "__main__":
+    data_folder = Path(__file__).absolute().parent
+    original_file = str(data_folder / f"original_test.json")
```
Comment (Kipok) [inline datasets/aime-2024/prepare.py:127]:
```suggestion
    original_file = str(data_folder / "original_test.json")
```

File: datasets/aime-2024/prepare.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    data_folder.mkdir(exist_ok=True)
+    output_file = str(data_folder / f"test.jsonl")
```
Comment (Kipok) [inline datasets/aime-2024/prepare.py:129]:
```suggestion
    output_file = str(data_folder / "test.jsonl")
```

File: datasets/aime-2024/prepare.py (hunk 1 above)
Comment (Kipok) [inline datasets/aime-2024/prepare.py:1]: Please add these two new test files to the tests/test_datasets.py

Comment (Kipok) [inline datasets/aime-2024/prepare.py:1]: Let's create a new requirements.txt inside datasets folder and put all necessary packages there and add it to the .github/workflow/tests.yml to both unit and cluster tests as a separate installation. Let's also add it to docs/prerequisites.md to say that to get all datasets you need to install those requirements both on cluster and locally. Besides requests/bs4 please also add pyyaml and `evalplus @ git+https://github.com/evalplus/evalplus@c91370f` to that req file

**Review by Kipok — CHANGES_REQUESTED** [review]

File: datasets/aime-2024/prepare.py (hunk 1 above)
Comment (wedu-nvidia) [reply datasets/aime-2024/prepare.py:1]: ok, i will add

Comment (wedu-nvidia) [reply datasets/aime-2024/prepare.py:1]: Added

Comment (wedu-nvidia) [reply datasets/aime-2024/prepare.py:16]: added

**Comment by Kipok** [comment]
Thanks, looks great now!

**Pull request merged**